src/devices/Ssd1351/Ssd1351.cs

```diff
@@ -0,0 +1,275 @@
+
+        /// <summary>
+        /// Send a command to the the display controller along with parameters.
+        /// </summary>
+        /// <param name="command">Command to send.</param>
+        /// <param name="data">Span to send as parameters for the command.</param>
+        private void SendCommand(Ssd1351Command command, Span<byte> data)
+        {
+            Span<byte> commandSpan = stackalloc byte[] { (byte) command };
+
+            SendSPI(commandSpan, true);
+
+            if (data != null && data.Length > 0)
+            {
+                SendSPI(data);
+
+                // detect certain commands that may alter the state of the device. This is done as the 
+                // SPI device cannot read registers from the ssd1351 and so changes need to be captured
+                switch (command)
+                {
+                    // capture changes to the colour depth and colour sequence
+                    case Ssd1351Command.SetRemap:
+                        _colorSequence = (ColorSequence)((data[0] >> 2) & 0x01);
+                        _colorDepth = (ColorDepth)((data[0] >> 6) & 0x03);
+                        break;
+                }
+
+            }
+
+        }
+
+        /// <summary>
+        /// Send data to the display controller.
+        /// </summary>
+        /// <param name="data">The data to send to the display controller.</param>
+        public void SendData(Span<byte> data)
+        {
+            if (data == null)
+            {
+                throw new ArgumentNullException(nameof(data));
+            }
+
+            // create a buffer to contain the data plus pseudo command to indicate data.
+            Span<byte> buffer = stackalloc byte[data.Length + 1];
+
+            buffer[0] = 0x40; // Control byte indicating that the following bytes are data and not command parameters.
+            data.CopyTo(buffer.Slice(1));
+            SendSPI(buffer);
+        }
+
+        /// <summary>
+        /// Write a block of data to the SPI device
+        /// </summary>
+        /// <param name="data">The data to be sent to the SPI device</param>
+        /// <param name="blnIsCommand">A flag indicating that the data is really a command when true or data when false.</param>
+        private void SendSPI(Span<byte> data, bool blnIsCommand = false)
+        {
+            int index = 0;
+            int len;
+
+            // set the DC pin to indicate if the data being sent to the display is DATA or COMMAND bytes.
+            _gpioDevice.Write(_dcPinId, blnIsCommand ? PinValue.Low : PinValue.High);
+
+            // write the array of bytes to the display. (in chunks of SPI Buffer Size)
+            do
+            {
+                // calculate the amount of spi data to send in this chunk
+                len = Math.Min(data.Length - index, _spiBufferSize);
+                // send the slice of data off set by the index and of length len.
+                _spiDevice.Write(data.Slice(index, len));
```

Comment:
when will this be more than one chunk?

Reply:
Much of the time. Full image would be 32K (128*128*2 bytes) bytes of data ignoring the command bytes. The default for the raspbian SPI buffer is 4K, I believe, and so unless you use a non standard buffer size then chunking the output is inevitable.